superset/db_engine_specs/hive.py

```diff
@@ -79,6 +79,12 @@ def upload_to_s3(filename: str, upload_prefix: str, table: Table) -> str:
         )
 
     s3 = boto3.client("s3")
+
+    # The location is merely an S3 prefix and thus we first need to ensure that there is
```

Comment:
I discovered this issue when testing locally, i.e., given that the location field is a directory as opposed to a file any file within said directory is slurped into the table.